### DISCUSSION TITLE
RAMJobStore limits on number of jobs

### DISCUSSION THREAD

**Discussion by kyletran200** [quartznet/quartznet#1642] [Q&A] [answered]
Hello.

I am currently working on a scheduling application that will use the RAMJobStore to store information on the jobs that were scheduled. First of all, does the RAMJobStore store the complete history of all jobs scheduled? If so, what's the best practice to check the amount of memory used by the RAMJobStore, and the memory used for each job object. This information would be useful in determining how many jobs our device can store in RAM and in developing a clean up policy for the RAMJobStore.

Thanks!

**Comment by lahma** [comment, marked as answer]
RAMJobStore doesn't hold statistics or job history. Just the current valid jobs and triggers. You could look at `LoggingJobHistoryPlugin` and `LoggingTriggerHistoryPlugin` for ideas on how to gather such info. You will probably need some persistence medium if you don't want to fill the memory though.
